### ISSUE TITLE
Define specific intended outputs of this repo

### ISSUE THREAD

**Issue by cjacoby** [ejhumphrey/minst-dataset#26]
AKA if you were to `make build` this repo, what *exactly* should it produce?

How do you envision using this with hcnn? Optimally, I would prefer to effectively separate the repos, so we don't have to make the HCNN repo depend on this one in any way, and it can just consume the .csv files generated.

What I would prefer / am thinking:
* minst-dataset generates the dataset to work with. This includes:
    * "note" audio files
    * a "master" dataset notes_index.csv containing all the metadata from all the datasets, pointing to the relevant notes.
    * train/test notes_index.csv's *for each* in ['rwc', 'philharmonia', 'uiowa']
    * Possibly a single json/yaml metadata file which points to all of these, which you can pass to the next program for training.

Another approach would be to do the test_set/folds generation in hcnn, and include/depend on this repo, but I think it would be simpler / more modular if we did it this way.

Thoughts?

Perhaps we can document this in the README when we're done?

(And I think I'm blocked on moving forward on the current state in HCNN on this discussion, because it completely affects how I consume this data).

**Comment by ejhumphrey** [comment]
Good question. Let's discuss.

To your list (in order):
- yes
- yes
- I would prefer partitions are lists of indices into the master note dataframe, rather than duplicating this information into additional tables.
- I don't think this is necessary if we do this right.

I don't mind using the `minst` library as a means of managing this data, ensuring it's integrity, etc. This is how @rabitt handles the MedleyDB library (https://github.com/marl/medleydb) and I think it's pretty shiny. No matter how you slice it, you're either using pandas or numpy or some library as an interface layer, might as well be one that can tell you the data is good. If anything, I think this upholds the ideal of modularity (tho not independence, which is maybe what you meant?).

Note that this is only because the audio data is dereferenced through the CSV file. Alternatively, we could save audio data in a big HDF5 / tarbal, making this all a little easier.

**Comment by ejhumphrey** [comment]
I think this is resolved now ... ? @cjacoby feel free to close or followup accordingly

**Comment by ejhumphrey** [comment]
this is the last issue blocking the alpha, are we clear to cut a proper release?

**Comment by cjacoby** [comment]
Running a build; I'll give the go-ahead if it works.

**Comment by cjacoby** [comment]
I have one more request for an output, but I'm going to make a separate issue for it.

**Comment by cjacoby** [comment]
My build seems to be successful, so I'm happy to accept alpha, maybe pending on #34.

**Comment by ejhumphrey** [comment]
is that something you can do or should I plan to do that tonight?

**Comment by cjacoby** [comment]
If it's all the same to you, i'd rather spend my cycles on harmonic-cnn
repo.

**Comment by ejhumphrey** [comment]
sounds good to me, I'll take a look at it and see what I can do.

**Comment by cjacoby** [comment]
Completed/closed with #35.
